lib/macho/load_commands.rb

```diff
@@ -400,6 +400,32 @@ def initialize(view, cmd, cmdsize, segname, vmaddr, vmsize, fileoff,
       @flags = flags
     end
 
+
+    # All sections referenced within this segment.
+    # @return [Array<MachO::Section>] if the Mach-O is 32-bit
+    # @return [Array<MachO::Section64>] if the Mach-O is 64-bit
+    def sections
+      section_klass = case self
+      when MachO::SegmentCommand64
+        MachO::Section64
+      when MachO::SegmentCommand
+        MachO::Section
+      end
+
+      sections = []
+      return sections if nsects.zero?
+
+      offset = view.offset + self.class.bytesize
+
+      nsects.times do
+        section_bin = view.raw_data[offset, section_klass.bytesize]
+        sections << section_klass.new_from_bin(view.endianness, section_bin)
+        offset += section_klass.bytesize
+      end
```

Comment:
I'll think I'll switch to this - getting rid of the `offset` variable is nice and that dynamic unpack is a nice touch :+1: